src/math/median.ts

```diff
@@ -0,0 +1,36 @@
+import { mean } from './mean';
+
+/**
+ * Calculates the median of an array of numbers.
+ *
+ * If the array is empty, this function returns `NaN`.
+ * If the array has an odd number of elements, it returns the middle element.
+ * If the array has an even number of elements, it returns the average of the two middle elements.
+ *
+ * @param {number[]} nums - An array of numbers to calculate the median.
+ * @returns {number} The median of all the numbers in the array.
+ *
+ * @example
+ * const arrayWithOddNumberOfElements = [1, 2, 3, 4, 5];
+ * const result = median(arrayWithOddNumberOfElements);
+ * // result will be 3
+ *
+ * @example
+ * const arrayWithEvenNumberOfElements = [1, 2, 3, 4];
+ * const result = median(arrayWithEvenNumberOfElements);
+ * // result will be 2.5
+ */
+export function median(nums: readonly number[]): number {
+  if (nums.length === 0) {
+    return NaN;
+  }
+
+  const sorted = [...nums].sort((a, b) => a - b);
+  const middleIndex = Math.floor(sorted.length / 2);
+
+  if (sorted.length % 2 === 0) {
+    return mean([sorted[middleIndex - 1], sorted[middleIndex]]);
```

Comment:
```suggestion
    return (sorted[middleIndex - 1] + sorted[middleIndex])/ 2;
```

Since we are simply getting the mean of two numbers, we might directly calculate the average here.